cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -30,21 +30,22 @@ namespace cudf::io::parquet::detail {
 
 void reader::impl::decode_page_data(size_t skip_rows, size_t num_rows)
 {
-  auto& chunks              = _pass_itm_data->chunks;
-  auto& pages               = _pass_itm_data->pages_info;
-  auto& page_nesting        = _pass_itm_data->page_nesting_info;
-  auto& page_nesting_decode = _pass_itm_data->page_nesting_decode_info;
+  auto& pass    = *_pass_itm_data;
+  auto& subpass = *pass.subpass;
```

Comment:
```suggestion
  auto& subpass = *pass.subpass;
  auto& chunks  = pass.chunks;
  auto& pages   = subpass.pages;
```
Would cut down on a lot of the diffs in this function, but if you want to emphasize the "pass" aspect then nevermind :)

Reply:
Yeah I want to be very explicit about where various data is coming from, pass or subpass.  Although in reality, the only thing that ever really changes is the list of pages (chunks are constant across an entire pass).